lib/stimulus_reflex.rb

```diff
@@ -20,5 +20,53 @@
 
 module StimulusReflex
   class Engine < Rails::Engine
+    NODE_VERSION_FORMAT = /(\d\.\d\.\d.*):/
+
+    initializer "stimulus_reflex.verify_caching_enabled" do
+      unless caching_enabled?
+        puts <<~WARN
+          Stimulus Reflex requires caching to be enabled. Caching allows the session to be modified during ActionCable requests.
+          To enable caching in development, run:
+
+            rails dev:cache
+        WARN
+        exit
+      end
+    end
+
+    initializer "stimulus_reflex.verify_npm_package_version" do
+      unless node_version_matches?
+        puts <<~WARN
+          The Stimulus Reflex javascript package version (#{node_package_version}) does not match the Rubygem version (#{gem_version}).
+          To update the Stimulus Reflex node module:
+
+            yarn upgrade stimulus_reflex@#{gem_version}
+        WARN
+        exit
+      end
+    end
+
+    private
+
+    def caching_enabled?
```

Comment:
I noticed you took careful steps to align the different „pre“ version formats between node and ruby. Just checking: you did test this, correct?

Reply:
@julianrubisch I did some tests. It looks good, we also did the same substitution when adjusting the install command, see https://github.com/hopsoft/stimulus_reflex/pull/297/commits/53643b5af16e3c9aa674cd26e7177aeedf5fbb95#diff-91900d59e187ebdd5ac9ec1ada32fe44L9

Reply:
Yep, it's on purpose to not confuse anyone (and it's the actual versions we compare, so a sanity check at the same time).